service/http/dffml_service_http/routes.py

```diff
@@ -156,16 +181,46 @@ async def multicomm_dataflow(self, config, request):
         inputs = []
         # If data was sent add those inputs
         if request.method == "POST":
-            # Accept a list of input data
-            # TODO validate that input data is dict of list of inputs each item
-            # has definition and value properties
-            for ctx, client_inputs in (await request.json()).items():
-                for input_data in client_inputs:
-                    if (
-                        not input_data["definition"]
-                        in config.dataflow.definitions
-                    ):
-                        return web.json_response(
+            # Accept a list of input data according to config.input_mode
+            if config.input_mode == "default":
+                # TODO validate that input data is dict of list of inputs each item
+                # has definition and value properties
+                for ctx, client_inputs in (await request.json()).items():
+                    for input_data in client_inputs:
+                        if (
+                            not input_data["definition"]
+                            in config.dataflow.definitions
+                        ):
+                            return web.json_response(
+                                {
+                                    "error": f"Missing definition for {input_data['definition']} in dataflow"
+                                },
+                                status=HTTPStatus.NOT_FOUND,
+                            )
+                    inputs.append(
+                        MemoryInputSet(
+                            MemoryInputSetConfig(
+                                ctx=StringInputSetContext(ctx),
+                                inputs=[
+                                    Input(
+                                        value=input_data["value"],
+                                        definition=config.dataflow.definitions[
+                                            input_data["definition"]
+                                        ],
+                                    )
+                                    for input_data in client_inputs
+                                ],
+                            )
+                        )
+                    )
+            elif config.input_mode.startswith("def:"):
+                # TODO
+                # add an option to parse input to json?
+                # or add `formatter` in config which points to dataflow
+                # for now :json loads
+                input_def = config.input_mode.replace("def:","")
```

Comment:
Remove `def:`, then split on ':' treat last index as definition and first index as preprocessing.

So,

- `def:bytes:mydefinition`

  - `Input(value=await response.read()`

- `def:str:mydefinition`

  - `Input(value=await response.text()`

- `def:json:mydefinition`

  - `Input(value=await response.json()`

- `def:stream:mydefinition`

  - `Input(value=response`